praw/models/reddit/rules.py

```diff
@@ -0,0 +1,296 @@
+
+    def _make_rule_list(self, data_list: List[Dict[str, str]]) -> List[Rule]:
+        """Convert a list of rule dicts to Rule objects.
+
+        :param data_list: The list of dictionaries
+        :returns: A list of instances of :class:`.Rule`.
+        """
+        return [self._make_rule(ruledata) for ruledata in data_list]
+
+    def add(
+        self,
+        short_name: str,
+        kind: str,
+        description: str = "",
+        violation_reason: Optional[str] = None,
+    ) -> Rule:
+        """Add a removal reason to this subreddit.
+
+        :param short_name: The name of the rule.
+        :param kind: The kind of item that the rule applies to. One of
+            ``link``, ``comment``, or ``all``.
+        :param description: The description for the rule. Optional.
+        :param violation_reason: The reason that is shown on the report menu.
+            If a violation reason is not specified, the short name will be used
+            as the violation reason.
+        :returns: The Rule added.
+
+        To add rule ``No spam`` to the subreddit ``NAME`` try:
+
+        .. code-block:: python
+
+           reddit.subreddit('NAME').rules.add(
+               "No spam",
+               "all",
+               description="Do not spam. Spam bad")
+
+        """
+        data = {
+            "r": str(self.subreddit),
+            "description": description,
+            "kind": kind,
+            "short_name": short_name,
+            "violation_reason": short_name
+            if violation_reason is None
+            else violation_reason,
+        }
+        request_data = self._reddit.request(
+            "POST", API_PATH["add_subreddit_rule"], data=data
+        )
+        self._reddit._objector.objectify(request_data)
+        # We want any errors to be passed, but we don't want to use the
+        # objected value because it returns a LiveThread.
+        return self._make_rule(loads(request_data["json"]["data"]["rules"])[0])
+
+    def reorder(self, rule_list: List[Rule]) -> List[Rule]:
+        """Reorder the rules of a subreddit.
+
+        :param rule_list: The list of rules, in the wanted order. Each index of
+            the list indicates the position of the rule.
+        :returns: A list containing the rules in the specified order.
+
+        For example, to move the fourth rule to the first position, and then to
+        move the prior first rule to where the third rule originally was:
+
+        .. code-block:: python
+
+            subreddit = reddit.subreddit("subreddit")
+            rules = list(subreddit.rules)
+            new_rules = rules[3:4] + rules[1:3] + rules[0:1] + rules[4:]
+            # NOTE: Adding an upper bound returns a list
+            # Alternate: [rules[3]] + rules[1:3] + [rules[0]] + rules[4:]
+            new_rule_list = subreddit.rules.reorder(new_rules)
+
+        """
+        order_string = quote(
+            ",".join([str(item) for item in rule_list]), safe=","
+        )
```

Comment:
Why do you need to call `quote`?

Reply:
Reddit urlencodes the spaces in rule names, but not the commas.

Reply:
Weird. And passing these as an array, for example, doesn't work? The `quote` is totally necessary? I ask just because normally it seems like prawcore takes care of any encoding that needs to be done, but I also know that the Reddit API is quirky so things like this happen.

Reply:
I don't understand how encoding by prawcore works. Furthermore, prawcore probably spaces the arguments, but in this case, there can't be any spaces between an argument.

Reply:
@PythonCoderAS can you give an example string that requires `quote` to be done here?